coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -57,6 +58,111 @@ public async Task<Product> GetProductAsync(int productId)
             return product;
         }
 
+        public Task<InitiateProductResponse> AddProduct(InitiateProductRequest newProduct, IEnumerable<UserGroup> allowedUserGroups)
+        {
+            var product = new Product()
+            {
+                Price = newProduct.Price,
+                Description = newProduct.Description,
+                Name = newProduct.Name,
+                NumberOfTickets = newProduct.NumberOfTickets,
+                Visible = newProduct.Visible
+            };
+            
+            var productUserGroups = allowedUserGroups.Select(userGroup => new ProductUserGroup
+            {
+                ProductId = product.Id,
+                UserGroup = userGroup
+            }).ToList();
+
+            _context.ProductUserGroups.AddRange(productUserGroups);
+            
+            _context.Products.Add(product);
+            
+            _context.SaveChanges();
+
+
+            var result = new InitiateProductResponse
+            {
+                Price = product.Price,
+                Description = product.Description,
+                Name = product.Name,
+                NumberOfTickets = product.NumberOfTickets,
+                Visible = product.Visible
+            };
+
+            return Task.FromResult(result);
+        }
+        
+        private Product GetProduct(int productId)
+        {
+            return _context.Products.Where(p => p.Id == productId).FirstOrDefault();
+        }
+        
+        public Task<InitiateProductResponse> UpdateProduct(InitiateProductRequest changedProduct)
+        {
+            var product = GetProduct(changedProduct.Id);
+
+            if (changedProduct.Id != default(int))
+            {
+                Log.Information($"Changing Id of product from {product.Id} to {changedProduct.Id}");
+                product.Id = changedProduct.Id;
+            }
+
+            if (changedProduct.NumberOfTickets != default(int))
+            {
+                Log.Information($"Changing NumberOfTickets of product from {product.NumberOfTickets} to {changedProduct.NumberOfTickets}");
+                product.NumberOfTickets = changedProduct.NumberOfTickets;
+            }
+
+            if (!string.IsNullOrEmpty(changedProduct.Name))
```

Comment:
This would be logged as the name being changed, even if it wasn't. We need to change this condition slightly